e2e-tests/scenarios_view_users.js

```diff
@@ -8,24 +8,34 @@ describe('rocketvoip', function () {
             var mockFunction = function () {
                 angular.module('httpBackendMock', ['ngMockE2E'])
                     .run(function ($httpBackend) {
+                        var testSipClients = [];
+                        var nextId = 0;
                         $httpBackend.whenGET(/\/sipclients/).respond(function () {
-                            return [200, []];
+                            return [200, testSipClients, {}];
                         });
                         $httpBackend.whenPOST(/\/sipclients/).respond(function (method, url, data) {
                             var sipClient = angular.fromJson(data);
-                            sipClient.id = Math.floor((Math.random() * 30000) + 100);
+                            sipClient.id = nextId;
+                            testSipClients.push(sipClient);
                             return [200, sipClient, {}];
                         });
                         $httpBackend.whenPUT(/\/sipclients\//).respond(function (method, url, data) {
                             var sipClient = angular.fromJson(data);
+                            angular.forEach(testSipClients, function (client, key) {
+                                if(client.id == sipClient.id) {
+                                    testSipClients[key] = sipClient;
+                                }
+                            });
                             return [200, sipClient, {}];
                         });
                         $httpBackend.whenDELETE(/\/sipclients\//).respond(function () {
+                            testSipClients.pop();
+                            testSipClients.pop();
```

Comment:
Why two times? 

`for (var i = array.length - 1; i >= 0; i--) {		
if (array[i].id == toDelete.id) {		
array.splice(i, 1);		
}		
}`